src/sinks/file.rs

```diff
@@ -0,0 +1,316 @@
+use crate::{
+    buffers::Acker,
+    event::Event,
+    sinks::util::{
+        encoding::BasicEncoding,
+        file::{EmbeddedFileSink, FileSink},
+        SinkExt,
+    },
+    template::Template,
+    topology::config::DataType,
+};
+
+use futures::{future, Async, AsyncSink, Sink, StartSend};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::time::Instant;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct FileSinkConfig {
+    #[serde(rename = "path")]
+    pub path_template: String,
+    #[serde(rename = "timeout", default = "default_close_timeout_secs")]
+    pub close_timeout_secs: Option<u64>,
+    pub encoding: Option<BasicEncoding>,
+}
+
+fn default_close_timeout_secs() -> Option<u64> {
+    None //means that we don't close files, so no overhead
+}
+
+impl FileSinkConfig {
+    pub fn new(path_template: String) -> Self {
+        Self {
+            path_template,
+            close_timeout_secs: default_close_timeout_secs(),
+            encoding: None,
+        }
+    }
+}
+
+#[typetag::serde(name = "file")]
+impl crate::topology::config::SinkConfig for FileSinkConfig {
+    fn build(&self, acker: Acker) -> Result<(super::RouterSink, super::Healthcheck), String> {
+        let sink = PartitionedFileSink::new(
+            Template::from(&self.path_template[..]),
+            self.close_timeout_secs,
+            self.encoding.clone(),
+        )
+        .stream_ack(acker);
+
+        Ok((Box::new(sink), Box::new(future::ok(()))))
```

Comment:
I assume we probably want to healthcheck that we will have permissions to create and append files in the specified path?

Reply:
@LucioFranco this is generally not possible since paths are templates.
Possible to check path for being strictly template or normal path and performing healthcheck in second case. Not sure whether it is overengineering or not..